src/modules/avformat/producer_avformat.c

```diff
@@ -1246,6 +1247,102 @@ static int pick_av_pixel_format( int *pix_fmt )
 	return 0;
 }
 
+struct sliced_pix_fmt_conv_t
+{
+	int width, height, slice_w;
+	AVFrame *frame;
+	AVPicture *output;
+	enum AVPixelFormat src_format, dst_format;
+	const AVPixFmtDescriptor *src_desc, *dst_desc;
+	int flags, src_colorspace, dst_colorspace, src_full_range, dst_full_range;
+};
+
+static int sliced_h_pix_fmt_conv_proc( int id, int idx, int jobs, void* cookie )
+{
+	uint8_t *out[4];
+	const uint8_t *in[4];
+	int in_stride[4], out_stride[4];
+	int src_v_chr_pos = -513, dst_v_chr_pos = -513, ret, i, slice_x, slice_w, w, h, mul, field, slices, interlaced = 0;
+
+	struct SwsContext *sws;
+	struct sliced_pix_fmt_conv_t* ctx = ( struct sliced_pix_fmt_conv_t* )cookie;
+
+	interlaced = ctx->frame->interlaced_frame;
+	field = ( interlaced ) ? ( idx & 1 ) : 0;
+	idx = ( interlaced ) ? ( idx / 2 ) : idx;
+	slices = ( interlaced ) ? ( jobs / 2 ) : jobs;
+	mul = ( interlaced ) ? 2 : 1;
+	h = ctx->height >> !!interlaced;
+	slice_w = ctx->slice_w;
+	slice_x = slice_w * idx;
+	slice_w = FFMIN( slice_w, ctx->width - slice_x );
+
+	if ( AV_PIX_FMT_YUV420P == ctx->src_format )
+		src_v_chr_pos = ( !interlaced ) ? 128 : ( !field ) ? 64 : 192;
+
+	if ( AV_PIX_FMT_YUV420P == ctx->dst_format )
+		dst_v_chr_pos = ( !interlaced ) ? 128 : ( !field ) ? 64 : 192;
+
+	mlt_log_debug( NULL, "%s:%d: [id=%d, idx=%d, jobs=%d], interlaced=%d, field=%d, slices=%d, mul=%d, h=%d, slice_w=%d, slice_x=%d ctx->src_desc=[log2_chroma_h=%d, log2_chroma_w=%d], src_v_chr_pos=%d, dst_v_chr_pos=%d\n",
+		__FUNCTION__, __LINE__, id, idx, jobs, interlaced, field, slices, mul, h, slice_w, slice_x, ctx->src_desc->log2_chroma_h, ctx->src_desc->log2_chroma_w, src_v_chr_pos, dst_v_chr_pos );
+
+	if ( slice_w <= 0 )
+		return 0;
+
+	sws = sws_alloc_context();
```

Comment:
I don't have a lot of experience with the SWS API. What is the cost of allocating a new context for each slice? Is it expensive?

Would it be possible to create a single context for each frame and then have each slice use the same context?

Would it be possible to create a single context on initialization and use the same context for all slices and all frames?